executor/executor.go

```diff
@@ -304,6 +305,68 @@ type ShowDDLJobsExec struct {
 	jobs   []*model.Job
 }
 
+// ShowDDLJobIDExec represent a show DDL jobs job_id executor.
+type ShowDDLJobIDExec struct {
+	baseExecutor
+
+	cursor int
+	jobs   []*model.Job
+	jobIDs []int64
+	query  string
+}
+
+// Open implements the Executor Open interface.
+func (e *ShowDDLJobIDExec) Open(ctx context.Context) error {
+	if err := e.baseExecutor.Open(ctx); err != nil {
+		return errors.Trace(err)
+	}
+	jobs, err := admin.GetDDLJobs(e.ctx.Txn())
+	if err != nil {
+		return errors.Trace(err)
+	}
+	historyJobs, err := admin.GetHistoryDDLJobs(e.ctx.Txn())
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	e.jobs = append(e.jobs, jobs...)
+	e.jobs = append(e.jobs, historyJobs...)
+
+	e.cursor = 0
+	return nil
+}
+
+// Next implements the Executor Next interface.
+func (e *ShowDDLJobIDExec) Next(ctx context.Context) (Row, error) {
+	if e.cursor >= len(e.jobs) {
+		return nil, nil
+	}
+
+	job := e.jobs[e.cursor]
+	row := types.MakeDatums(job.Query)
+	e.cursor++
+
+	return row, nil
+}
+
+// NextChunk implements the Executor NextChunk interface.
+func (e *ShowDDLJobIDExec) NextChunk(ctx context.Context, chk *chunk.Chunk) error {
+	chk.Reset()
+	if e.cursor >= len(e.jobs) {
+		return nil
+	}
+	numCurBatch := mathutil.Min(e.maxChunkSize, len(e.jobs)-e.cursor)
+	for i := e.cursor; i < e.cursor+numCurBatch; i++ {
+		for j := 0; j < len(e.jobIDs); j++ {
+			if e.jobIDs[j] == e.jobs[i].ID {
+				chk.AppendString(0, e.jobs[i].Query)
```

Comment:
Could we put this logical to `ShowDDLJobs` directly?
If can't, please add tests in parser_test.go and executor_test.go.

Reply:
ok I added the test

Reply:
Why we can't put this logical to `ShowDDLJobs ` directly?
And the name of `AdminShowDDLJobID` is a bit difficult to understand.

Reply:
because using admin show ddl jobs  display is too hard to read.

Reply:
Got it.
s/AdminShowDDLJobID/AdminShowDDLJobQuerys?
s/admin show ddl job/admin show ddl job querys ?

Reply:
Please address comments. @ciscoxll